### PR TITLE
feat(fileIO): cache SandboxedInstance.

### PR DESCRIPTION
**Pull request by pbailly**
Side effect of the Daikon modification[1].
Modifications on RuntimeInfo are mandatory.
The BeamLocalRunnerOption is an optimization allowed by the caching.
**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [x] Tests for the changes have been added (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
[1] https://github.com/Talend/daikon/pull/124

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/549/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/549/)

File: components-api/src/main/java/org/talend/components/api/component/runtime/JarRuntimeInfo.java [hunk 1]
```diff
@@ -115,4 +115,28 @@ public String getRuntimeClassName() {
...
+            return false;
+        }
+        JarRuntimeInfo other = (JarRuntimeInfo) obj;
+        return this.runtimeClassName.equals(other.runtimeClassName) && this.jarUrl.equals(other.jarUrl)
+                && this.depTxtPath.equals(other.depTxtPath);
+    }
+
+    @Override
+    public int hashCode() {
+        return toString().hashCode();
```
Comment (sgandon) [inline components-api/src/main/java/org/talend/components/api/component/runtime/JarRuntimeInfo.java]: this is quite dangerous to base the hash on the toString(), cause the toString may be altered without paying attention to the hash as it is not usually used for hashing.

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/576/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/576/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/578/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/578/)

**Pull request merged**